contracts/TreasuryExtender.sol

```diff
@@ -0,0 +1,464 @@
+     * @dev
+     *  Our first Allocator is at index 1, NOTE: 0 is a placeholder.
+     * @param id the id of the allocator, NOTE: valid interval: 1 =< id < allocators.length
+     * @return allocatorAddress the allocator's address
+     */
+    function getAllocatorByID(uint256 id) external view override returns (address allocatorAddress) {
+        allocatorAddress = address(allocators[id]);
+    }
+
+    /**
+     * @notice
+     *  Get the total number of Allocators ever registered.
+     * @dev
+     *  Our first Allocator is at index 1, 0 is a placeholder.
+     * @return total number of allocators ever registered
+     */
+    function getTotalAllocatorCount() external view returns (uint256) {
+        return allocators.length;
+    }
+
+    /**
+     * @notice
+     *  Get an Allocators limits.
+     * @dev
+     *  For an explanation of AllocatorLimits, see `setAllocatorLimits`
+     * @return the Allocator's limits
+     */
+    function getAllocatorLimits(uint256 id) external view override returns (AllocatorLimits memory) {
+        return allocatorData[allocators[id]][id].limits;
+    }
+
+    /**
+     * @notice
+     *  Get an Allocators performance.
+     * @dev
+     *  An Allocator's performance is the amount of `gain` and `loss` it has sustained in its
+     *  lifetime. `gain` is the amount of allocated tokens (underlying) acquired, while
+     *  `loss` is the amount lost. `gain` and `loss` are incremented separately.
+     *  Thus, overall performance can be gauged as gain - loss
+     * @return the Allocator's performance
+     */
+    function getAllocatorPerformance(uint256 id) external view override returns (AllocatorPerformance memory) {
+        return allocatorData[allocators[id]][id].performance;
+    }
+
+    /**
+     * @notice
+     *  Get an Allocators amount allocated.
+     * @dev
+     *  This is simply the amount of `token` which was allocated to the allocator.
+     * @return the Allocator's amount allocated
+     */
+    function getAllocatorAllocated(uint256 id) external view override returns (uint256) {
+        return allocatorData[allocators[id]][id].holdings.allocated;
+    }
+
+    /**
+     * @notice
+     *  Returns rewards from an Allocator to the Treasury.
+     * @dev
+     *  External hook: Logic is handled in the internal function.
+     *  Can only be called by the Guardian.
+     *
+     *  The assumption is that the reward tokens being withdrawn are going to be
+     *  either deposited into the contract OR harvested into allocated (underlying).
+     *
+     *  For this reason we don't need anything other than `balanceOf`.
+     * @param allocator the Allocator to returns rewards from
+     * @param token the reward token to withdraw
+     * @param amount the amount of the reward token to withdraw
+     */
+    function _returnRewardsToTreasury(
+        IAllocator allocator,
+        IERC20 token,
+        uint256 amount
+    ) internal onlyGuardian {
```

Comment:
shouldn't these be _onlyGuardian() since they use OlympusAccessControlledV2

Reply:
I changed the code style because it was suggested in the audit.